accepted/2021/architecture-targeting.md

```diff
@@ -0,0 +1,492 @@
+
+### Enable all build types for all (appropriate) verbs
+
+There are three build types that were defined earlier. All three are possible to create with `dotnet publish` but the same is not true for `dotnet build`. `dotnet build` does not enable creating a RID-specific framework-dependent app (w/o relying on MSBuild properties). Going forward, we need to ensure that both `build` and `publish` have the same capabilities for all three build types.
+
+This behavior largely follows from the previous sections, but good to call out.
+
+Up until now, `dotnet build` and `dotnet publish` have had slightly different behaviors and capabilities with respect to the topics discussed in this document. Going forward, they need to have the same behavior. There is some discussion of deprecating `dotnet publish`. That's out of scope of this doc. That said, a first step to deprecating `dotnet publish` would be rationalizing the behavior differences. Making `dotnet build` and `dotnet publish` expose the same model and provide the same behavior for RID, architecture and deployment targeting would be a great approach to that.
+
+Some people have asked that `dotnet publish` produces `-c release` apps by default. Great idea but out of scope for this document.
+
+### Enable RID targeting for all (appropriate) verbs
+
+Today, the CLI assumes that the implicit RID is the correct one to use for apphost creation, but is [wrong in some scenarios](https://github.com/dotnet/sdk/issues/17241) and does not provide an override.
+
+The primary scenario is enabling the native-architecture SDK to target an emulated architecture. For example, `dotnet run`, `dotnet test`, and `dotnet tool install` should all accept a RID to influence apphost creation, such as with the following commands:
+
+- `dotnet run -r osx-x64 --`.
+- `dotnet test -r win-x64`
+- `dotnet tool install -r win-x64`
+
+Note: `dotnet watch` is effectively a wrapper over `dotnet run` and `dotnet test` so would follow the same model as they do.
+
+Where appropriate, the SDK should encode a different implicit RID, such as `osx-x64` when targeting .NET Core 3.1 on Apple Silicon macOS machines with the Arm64 .NET 6 SDK. The SDK should understand that .NET Core 3.1 only supports x64 and not Arm64 for macOS. The x64 SDK should provide the same behavior in reverse.
+
+Tool installation with `dotnet tool install` is a particularly problematic scenario. At the time .NET 6 is released, one can expect that close to 100% of .NET tools (packaged via NuGet) will target a verion prior to .NET 6 and therefore be incapable on running on Arm64 (at least on macOS). However, when installed, the SDK will generate an apphost that matches the SDK RID. As a result, the tool will consistently fail to launch (since the required runtime will be missing). The SDK needs to apply the same implicit RID rules for .NET tools that it will apply for `dotnet build`. In addition, `dotnet tool install` should enable explicit RID targeting.
```

Comment:
```suggestion
Tool installation with `dotnet tool install` is a particularly problematic scenario. At the time .NET 6 is released, one can expect that close to 100% of .NET tools (packaged via NuGet) will target a version prior to .NET 6 and therefore be incapable on running on Arm64 (at least on macOS). However, when installed, the SDK will generate an apphost that matches the SDK RID. As a result, the tool will consistently fail to launch (since the required runtime will be missing). The SDK needs to apply the same implicit RID rules for .NET tools that it will apply for `dotnet build`. In addition, `dotnet tool install` should enable explicit RID targeting.
```
Typo: verion